pkg/virt-handler/vm.go

```diff
@@ -475,16 +478,24 @@ func (d *VirtualMachineController) closeLauncherClient(vmi *v1.VirtualMachineIns
 	d.launcherClientLock.Lock()
 	defer d.launcherClientLock.Unlock()
 
-	sockFile := cmdclient.SocketFromUID(d.virtShareDir, string(vmi.GetUID()))
+	vmiUUID, ok := d.vmisUUIDs[vmi.Name]
```

Comment:
to make this unique, both namespace and name will need to be used for the key.  so something like "namespace_name"